app/services/teacher_training_public_api/sync_provider.rb

```diff
@@ -5,11 +5,14 @@ def initialize(provider_from_api:, recruitment_cycle_year:)
       @recruitment_cycle_year = recruitment_cycle_year
     end
 
-    def call(run_in_background: true)
-      provider = create_or_update_provider(
-        provider_attrs_from(@provider_from_api),
-      )
-
+    def call(run_in_background: true, force_sync_courses: false)
```

Comment:
Im wondering about the need to force syncing courses at this stage, since this is already set here: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4084/files#diff-f48eb2c3a6563c8b75c612393c0862a248b03c7276ae21832337476c1f688317R63
Let me know your thoughts

Reply:
It's needed here because `TeacherTrainingPublicApi::SyncProvider` uses the `sync_courses` value to decided whether to sync courses itself (via `TeachTrainingPublicApi::SyncCourses` and its own network of classes). Ordinarily, `sync_courses` is set per provider as we keep migrating training/ratifying providers to our system. But in this context, it needs to be forced.

`FindSync::SyncProviderFromFind` and its related classes, from what I understand, are on their way out and will be removed when the API transition is complete. Setting `sync_courses` here is too late for any of the `TeacherTrainingPublicApi` updates.

Reply:
Ah I missed that bit in this class🤦  It seems though we have a variable that isn't being set anywhere and might have been left behind: https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/services/teacher_training_public_api/sync_provider.rb#L29 maybe we can tidy up the logic there with the new flag being passed in i.e. either set it on initialising this class and removing the force flag, or just removing this left over variable?

Reply:
Either of those options sounds like a good improvement

Reply:
Yes, I'll remove `@sync_courses`, it's no longer needed.

I think think setting `sync_courses` to `true` for each provider is better in the context of local dev data, as these courses would only be there if `sync_courses` was set to `true` in the first place, therefore closer to reality.

Reply:
Then again, `sync_courses` needs to be set to something when creating a provider. I've merged the logic of the force flag and the instance variable so that the value of the force flag is used as `sync_courses:` when creating new providers but may also be used to import courses for existing providers with `sync_courses: false`, if needed.